File: src/main/java/cum/jesus/jesusclient/injection/mixins/minecraft/client/gui/GuiScreen.java
package cum.jesus.jesusclient.injection.mixins.minecraft.client.gui;

import cum.jesus.jesusclient.JesusClient;
import net.minecraft.client.Minecraft;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.Shadow;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;

@Mixin(net.minecraft.client.gui.GuiScreen.class)
@SideOnly(Side.CLIENT)
public final class GuiScreen {
    @Shadow
    public Minecraft mc;

    @Inject(method = "sendChatMessage(Ljava/lang/String;Z)V", at = @At("HEAD"), cancellable = true)
    private void sendChatMessage(String msg, boolean addToChat, CallbackInfo ci) {
        if (msg.startsWith("-") && msg.length() > 1) {
            if (JesusClient.instance.commandHandler.execute(msg)) {
                mc.ingameGUI.getChatGUI().addToSentMessages(msg);
            }

            ci.cancel();
        }
    }
}

File: src/main/kotlin/cum/jesus/jesusclient/injection/plugin/guiscreen.kt
package cum.jesus.jesusclient.injection.plugin

import cum.jesus.jesusclient.JesusClient
import cum.jesus.jesusclient.module.modules.render.ClickGUIModule
import dev.falsehonesty.asmhelper.dsl.At
import dev.falsehonesty.asmhelper.dsl.InjectionPoint
import dev.falsehonesty.asmhelper.dsl.code.CodeBlock
import dev.falsehonesty.asmhelper.dsl.inject
import net.minecraft.client.Minecraft

fun injectGuiScreen() {
    injectSendChatMessage()
}

fun injectSendChatMessage() = inject {
    className = "net/minecraft/client/gui/GuiScreen"
    methodName = "sendChatMessage"
    methodDesc = "(Ljava/lang/String;Z)V"

    at = At(InjectionPoint.HEAD)

    codeBlock {
        val mc = shadowField<Minecraft>()

        val local1 = shadowLocal<String>()

        code {
            if (JesusClient.isLoaded() && local1.startsWith(ClickGUIModule.INSTANCE.commandPrefix.value) && local1.length > ClickGUIModule.INSTANCE.commandPrefix.value.length) {
                if (JesusClient.instance.commandHandler.execute(local1)) {
                    mc.ingameGUI.chatGUI.addToSentMessages(local1)
                }

                CodeBlock.methodReturn()
            }
        }
    }
}